cmd/appliance/upgrade/complete.go

```diff
@@ -112,7 +112,7 @@ func NewUpgradeCompleteCmd(f *factory.Factory) *cobra.Command {
 	flags.BoolVarP(&opts.backup, "backup", "b", opts.backup, "Backup primary Controller before completing the upgrade")
 	flags.StringVar(&opts.backupDestination, "backup-destination", "$HOME/Downloads/appgate/backup", "Specify path to download backup")
 	flags.StringVar(&opts.actualHostname, "actual-hostname", "", "If the actual hostname is different from that which you are connecting to the appliance admin API, this flag can be used for setting the actual hostname")
-	flags.IntVar(&opts.batchSize, "batch-size", 2, "Number of batch groups")
+	flags.IntVar(&opts.maxUnavailable, "max-unavailable", 1, "Defines how many gateways and logforwarders that are allowed to be upgraded per site at once. Setting this to a higher number will calculate batches according to the value set in this flag. Setting this to a higher value would make the upgrade process shorter at the cost of collective performance for users.")
```

Comment:
I think I am missing the "per site piece" and that is what the gatewayA1 v. gatewayB1 string differences are? because the sites are "A" and "B" in this test case?

Reply:
Yes, that's correct. The `--max-unavailable` is gateways and logforwarders unavailable _per site_, and the sites are reflected in the names. In this particular test, the max-unavailable option is set to 2 (see the args), so that means 2 gateways and logforwarders are allowed to be upgraded simultaneously.

Reply:
Is there any kind of "guard" here against having no HA whatsoever.
E.g. if you set "max unavailable" to 5 and a site has 3 gateways, will it update all 3 at a time or still least keep 1 running?

Reply:
Nope, there's no guard for it, but should we really have one? If the user thinks it's ok to have downtime during upgrades, they should be allowed to have it, no?

Reply:
Yeah makes sense, the default is 1, if they change it its up to them.